tests/ui/const-generics/generic_const_exprs/post-analysis-user-facing-param-env.rs

```diff
@@ -11,7 +11,6 @@ where
 {
     fn unimplemented(self, _: &Foo) -> Self::Output {
         //~^ ERROR method `unimplemented` is not a member of trait `std::ops::Add`
-        //~| ERROR type annotations needed
```

Comment:
There are some added `?` on the new `Result<DefId, ErrorGuaranteed>` so that seems to have caused this.